src/wp-content/themes/twentyfifteen/js/view-transitions.js

```diff
@@ -0,0 +1,97 @@
+if ( !! window.navigation && 'CSSViewTransitionRule' in window ) {
+	const getParentElement = ( element, selector ) => {
+		element = element.parentElement;
+		while ( element && element !== document ) {
+			if ( ! selector || element.matches( selector ) ) {
+				return element;
+			}
+		}
+		return null;
+	};
```

Comment:
@felixarntz: I’ve narrowed the problem you are having down to this `getClosestParent` function: it gets stuck in a loop because `element` remains the same while inside loop.

The fix:

```suggestion
	const getParentElement = ( element, selector ) => {
		element = element.parentElement;
		while ( element && element !== document ) {
			if ( ! selector || element.matches( selector ) ) {
				return element;
			}
			element = element.parentElement;
		}
		return null;
	};
```

Furthermore I think you could replace the functionality entirely with [`Element.closest()`](https://developer.mozilla.org/en-US/docs/Web/API/Element/closest)

Reply:
Yikes - of course it had to be an oversight like this 🤦 

Thanks! Great point on using `closest`, I've updated this in https://github.com/WordPress/wordpress-develop/pull/8131/commits/ab9b25b3d7e53b622f4b8847ad1774f551099fbd - works like a charm now!